### PR TITLE
Sign ARM binaries in pipeline

### PR DESCRIPTION
**Pull request by wtgodbe**
Was having difficulties getting https://github.com/dotnet/core-setup/pull/2383 to rebase properly, so here's a dup of that change-set, up-to-date with the latest changes to pipebuild. CC @eerhardt @chcosta @gkhanna79

### CODE REVIEW

**Comment by gkhanna79** [comment]
@dagood PTAL

File: buildpipeline/Core-Setup-Windows-Arm-BT.json [hunk 1]
```diff
@@ -136,141 +136,15 @@
...
-      "displayName": "Package",
-      "timeoutInMinutes": 0,
-      "task": {
         "id": "c6c4c611-aa2e-4a33-b606-5eaba2196824",
         "versionSpec": "1.*",
         "definitionType": "task"
       },
       "inputs": {
-        "solution": "$(PB_SourcesDirectory)\\src\\pkg\\packaging\\dir.proj",
+        "solution": "$(PB_SourcesDirectory)/src/src.builds",
```
Comment (dagood) [inline buildpipeline/Core-Setup-Windows-Arm-BT.json]: \<nit> Should these be backslashes to keep to convention?

**Comment by chcosta** [comment]
I've merged my changes into this PR.  @dagood @gkhanna79 @eerhardt , can you PTAL at the commit "Sign MSI before bundling" in this PR?

File: buildpipeline/Core-Setup-Windows-BT.json [hunk 2]
```diff
@@ -225,35 +656,44 @@
...
-      "value": "--branch $(PB_Branch) https://$(PB_VsoAccountName):$(PB_VsoPassword)@devdiv.visualstudio.com/DevDiv/_git/DotNet-Core-Setup-Trusted"
-    },
     "PB_VsoAccountName": {
       "value": "dn-bot"
     },
     "PB_VsoPassword": {
       "value": null,
       "isSecret": true
     },
+    "PB_Branch": {
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-Windows-BT.json]: We should just remove this variable.  It is useless - it doesn't do anything and shouldn't be used.

Comment (eerhardt) [inline buildpipeline/Core-Setup-Windows-BT.json]: Should use $(SourceBranch) instead.  See [https://github.com/dotnet/core-setup/pull/2460/files](https://github.com/dotnet/core-setup/pull/2460/files) for how it doesn't work.

---
In reply to: [118108233](https://github.com/dotnet/core-setup/pull/2477#discussion_r118108233) [](ancestors = 118108233)

Comment (chcosta) [inline buildpipeline/Core-Setup-Windows-BT.json]: It's used when creating the repo clone url -
```
PB_VsoRepoUrl": {

      "value": "--branch $(PB_Branch) https://$(PB_VsoAccountName):$(PB_VsoPassword)@devdiv.visualstudio.com/DevDiv/_git/DotNet-Core-Setup-Trusted"

    },
```

Comment (eerhardt) [inline buildpipeline/Core-Setup-Windows-BT.json]: It's not the right value though.  It doesn't get passed in correctly - so it will always be `master` even if you are on another branch.

Comment (chcosta) [inline buildpipeline/Core-Setup-Windows-BT.json]: Ah, I see where you're coming from.  `SourceBranch` is used to control VSTS' branch cloning, and `PB_SourceBranch` is currently used for our own clone command.  For the most part, we ignore `SourceBranch` since otherwise you have to pay extra careful attention to ensure that PB_SourceBranch and Branch align and the clone url + git's repository align.

Comment (eerhardt) [inline buildpipeline/Core-Setup-Windows-BT.json]: See the "Run PipeBuild.exe" step of https://devdiv.visualstudio.com/DevDiv/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=6161 

@weshaggard actually removed the `PB_Branch` variable completely last week.

Comment (chcosta) [inline buildpipeline/Core-Setup-Windows-BT.json]: > It's not the right value though. It doesn't get passed in correctly - so it will always be master even if you are on another branch.

That may be true, I haven't looked at what is actually being passed from PipeBuild recently.  I know that, for my own private builds, I quite often set PB_SourceBranch to a different value for validation.

Comment (chcosta) [inline buildpipeline/Core-Setup-Windows-BT.json]: I don't have a strong attachment to it, I was just pointing out what it was previously used for. I think our quick conversation cycle on this PR seems to be causing a bit of static.

I agree with cleaning this up, I think that should be a separate PR though.  I just updated the value in this PR because it was clearly wrong.

**Comment by wtgodbe** [comment]
@chcosta is this ready to merge?

**Comment by chcosta** [comment]
If @eerhardt doesn't have any other concerns, then yes.  I agree with his sentiment about the branch variables, but I wouldn't block on it.

**Comment by eerhardt** [comment]
Nope- no other concerns.

**Comment by chcosta** [comment]
@wtgodbe , are you planning to ask shiproom / port this to release/2.0.0 after the next official build validates the changes?

**Comment by wtgodbe** [comment]
@chcosta yes

**Comment by gkhanna79** [comment]
Lets get this change into 2.0.0 today.
